tests/drivers/build_all/sensor/i2c.dtsi

```diff
@@ -27,507 +27,434 @@
 #include <zephyr/dt-bindings/sensor/tmp11x.h>
 #include <zephyr/dt-bindings/sensor/tmp114.h>
 
-/****************************************
- * PLEASE KEEP REG ADDRESSES SEQUENTIAL *
- **************************************
- */
+#define TEST_I2C_DEVICE_WITH_ADDR(label, node, addr, ...) \
+	label: node@addr { \
+		reg = <addr>; \
```

Comment:
I believe this is needed to make sure node name address and reg value match even if we will be skipping values as we are not converting the input to MAKE_HEX to hex string

```suggestion
#define MAKE_HEX(addr) 0x##addr

#define TEST_I2C_DEVICE_WITH_ADDR(label, node, addr, ...) \
	label: node@addr { \
	reg = <MAKE_HEX(addr)>; \
	__VA_ARGS__ \
	}; 
```